### ISSUE TITLE
BUG: i cant pate images to excalidraw !

### ISSUE THREAD

**Issue by dumbdreamed** [zsviczian/obsidian-excalidraw-plugin#1013]
hi,thanks for all of the efforts, my problem is that when I want to paste some images into excalidraw canvas it gives me this error
![image](https://user-images.githubusercontent.com/64626106/218252265-ae7e22a6-6d29-40b8-9e7b-1a7d8e00f4e6.png)
and afterIi switch to other files the image that I have pasted before gets removed as nothing happened.

**Comment by zsviczian** [comment]
can you give me a bit more context? where are you pasting images from? What steps are you taking?
Can you share a quick screen capture so I can see what is going on?

**Comment by dumbdreamed** [comment]
hello, my friend thanks for answering, and thanks for your beautiful plugin!
this is a clip from the error that I get
I tried reinstalling excalidraw downgrading it, downgrading and even reinstalling the obsidian itself but nothing worked.
https://user-images.githubusercontent.com/64626106/218315580-322cba2a-f814-44cb-9897-a753d6eef1c1.mp4

**Comment by zsviczian** [comment]
And this happens with all your drawings all only on this one?
If only this one, then can you send me the excalidraw file it happens with so I can take a look.
If it happens with all your drawings then please send me the result of running Command Palette: Show Debug Info

**Comment by dumbdreamed** [comment]
thanks again,it actually happens with all of my excalidraw files when I try to paste something into them.
here is the result of the show debugging command :
SYSTEM INFO:
	Obsidian version: v1.1.9
	Installer version: v1.1.9
	Operating system: Windows 10 Enterprise 10.0.22623
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: Obsidianite v0.0.0
	Snippets enabled: 1
	Restricted mode: off
	Plugins installed: 44
	Plugins enabled: 39
		1: RTL Support v1.1.0
		2: Advanced Appearance v1.1.0
		3: Markdown Formatting Assistant v0.4.1
		4: Prettier Format v0.1.0
		5: Kanban v1.5.1
		6: Mind Map v1.1.0
		7: Folder Note v0.7.3
		8: Icon Folder v1.6.0
		9: File Explorer Note Count v1.2.0
		10: Aut-O-Backups v2.1.0
		11: Calendar v1.5.10
		12: Icons v0.3.0
		13: Macros v0.1.0
		14: Image Toolkit v1.3.1
		15: Highlightr v1.2.2
		16: Minimal Theme Settings v6.2.0
		17: Checklist v2.2.12
		18: Periodic Notes v0.0.17
		19: Relative Line Numbers v2.0.1
		20: Dataview v0.5.55
		21: Outliner v4.2.1
		22: Ozan's Image in Editor Plugin v2.1.6
		23: cMenu v1.1.2
		24: QuickAdd v0.9.1
		25: Templater v1.16.0
		26: Customizable Page Header and Title Bar v4.6.2
		27: ePub Reader v1.0.2
		28: Obsidian_to_Anki v3.4.1
		29: Style Settings v1.0.2
		30: PodNotes v2.8.0
		31: Tasks v1.25.0
		32: Tag Wrangler v0.5.6
		33: Annotator v0.2.10
		34: Obsidian42 - Strange New Worlds v1.1.2
		35: Recent Files v1.3.5
		36: Obsidian Trello v1.6.1
		37: MetaEdit v1.7.2
		38: CardBoard v0.7.2
		39: Excalidraw v1.8.12

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

**Comment by zsviczian** [comment]
After disabling plugins you need to restart Obsidian, some of the plugins do not clean up correctly.
Can you try and reproduce this problem in an empty test Vault with only Excalidraw installed?
...and pasting the image to other locations like into an Obsidian document works without any problems, correct?

**Comment by toxfu** [comment]
I have the same problem. Also when I try to open the obsidian markdown view it frezzes for some time

**Comment by dumbdreamed** [comment]
> After disabling plugins you need to restart Obsidian, some of the plugins do not clean up correctly. Can you try and reproduce this problem in an empty test Vault with only Excalidraw installed? ...and pasting the image to other locations like into an Obsidian document works without any problems, correct?

yes I restarted the obsidian after disabling all of the other add ons but it didn't fix the problem, i created a test vault and I only installed the excalidraw and i didn't get any error .

**Comment by LorinczAdrien** [comment]
Hi, I had the same issues as described above: Pasting image → Saving would lead to error.
I followed the steps proposed by @zsviczian, and after some experimenting I found out that the error was because of another plugin that changes how attachments are imported https://github.com/alberti42/obsidian-import-attachments-plus.
Once I disabled this plugin, everything worked again.

It may be reproducible with other plugins as well, that change how files are imported.
@zsviczian I know this is not a "bug" as per se, as the plugin is changing how Obsidian saves, but maybe a workaround could be implemented for this? I think interaction with these plugins can be quite common, and it is a hard issue to debug, solving it should avoid a lot of confusion.

**Comment by zsviczian** [comment]
Thanks. This is very helpful.
For simple selfish reasons (and to keep my motivation for working on this free/passion project plugin)... I strictly only work on integration features for plugins the I use. That is an extremely short list. Given Excalidraw is the most downloaded plugin in Community Plugins, I'd rather expect other plugins to test their features to fit with Excalidraw, not the other way around. Please raise an FR with https://github.com/alberti42/obsidian-import-attachments-plus

**Comment by LorinczAdrien** [comment]
> Thanks. This is very helpful. For simple selfish reasons (and to keep my motivation for working on this free/passion project plugin)... I strictly only work on integration features for plugins the I use. That is an extremely short list. Given Excalidraw is the most downloaded plugin in Community Plugins, I'd rather expect other plugins to test their features to fit with Excalidraw, not the other way around. Please raise an FR with https://github.com/alberti42/obsidian-import-attachments-plus

This makes total sense, and I wouldn't call it selfish, rather in some sense necessary. 
I have created an [issue on the Import Attachments+ repository](https://github.com/alberti42/obsidian-import-attachments-plus/issues/25).
@zsviczian After a quick glance in this repos' documentation, I can't find a list of plugins that are "officially" supported. Do you maintain a list of these, or is it just the ones you use day to day?

**Comment by zsviczian** [comment]
@LorinczAdrien, the best way to drive compatibility is by contacting the developers of the plugins you use and asking them to work on compatibility with Excalidraw. If plugin developers contact me, I am more than happy to cooperate to find sustainable solutions, like with the developer of PDF++, but I am not spending time unilaterally trying to track or build compatibility with others. As said above, simply not possible.

**Comment by LorinczAdrien** [comment]
@zsviczian You already answered this above, and I agreed. I was just asking if you maintain a list of plugins that "officially" work with Excalidraw?
